cmd/dockerd/daemon.go

```diff
@@ -277,6 +286,7 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 	errAPI := <-serveAPIWait
 	c.Cleanup()
 	shutdownDaemon(d)
+	close(watchStream)
```

Comment:
@aaronlehmann It should take a context then. Or the `cluster.Cleanup()` should close it after it knows there are no writes coming.